BMM.Core/ViewModels/Parameters/TrackCollectionParameter.cs

```diff
@@ -0,0 +1,21 @@
+using BMM.Core.ViewModels.Parameters.Interface;
+
+namespace BMM.Core.ViewModels.Parameters
+{
+    public class TrackCollectionParameter : ITrackCollectionParameter
+    {
+        public TrackCollectionParameter(
+            int? trackCollectionId = null,
```

Comment:
by moving the SharingSecret into it's own Parameter the trackCollectionId doesn't need to be nullable.